checks/evaluation/permissions/permissions.go

```diff
@@ -38,6 +38,58 @@ var (
 	topNoWriteID   = "gitHubWorkflowPermissionsTopNoWrite"
 )
 
+type PermissionLevel int
+
+const (
+	// PermissionLevelNone is a permission set to `none`.
+	PermissionLevelNone PermissionLevel = iota
+	// PermissionLevelRead is a permission set to `read`.
+	PermissionLevelRead
+	// PermissionLevelUnknown is for other kinds of alerts, mostly to support debug messages.
+	// TODO: remove it once we have implemented severity (#1874).
+	PermissionLevelUnknown
+	// PermissionLevelUndeclared is an undeclared permission.
+	PermissionLevelUndeclared
+	// PermissionLevelWrite is a permission set to `write` for a permission we consider potentially dangerous.
+	PermissionLevelWrite
+)
+
+// PermissionLocation represents a declaration type.
+type LocationType int
+
+const (
+	// PermissionLocationNil is in case the permission is nil.
+	PermissionLocationNil LocationType = iota
+	// PermissionLocationNotDeclared is for undeclared permission.
+	PermissionLocationNotDeclared
```

Comment:
I have added both to accommodate to this part:

https://github.com/ossf/scorecard/blob/2ef20f17fb2e64147c83440cd2c769653454015a/checks/evaluation/permissions/permissions.go#L120-L131

Here, `r.LocationType == nil` should return inconclusive score, `PermissionLocationTop` should log at warning level and other cases should debug. In my changes, I therefore need a `PermissionLocationType` that is not `PermissionLocationNil` so I can debug those instead of returning inconclusive score. This happens at these lines:

https://github.com/ossf/scorecard/blob/f38d5f3521caf14c1a26a8c12dfb93ae3efec668/checks/evaluation/permissions/permissions.go#L317-L329